sidecar/supervisor/appsupervisor.go

```diff
@@ -77,6 +98,10 @@ func (a *AppSupervisor) DoAppSupervision(agent *register.RegistrationAgent) {
 		select {
 		case sig := <-sigChan:
 			log.Infof("Intercepted signal '%s'", sig)
+
+			// forwarding signal to application parent process
+			exit(append(a.Helpers, a.App), sig)
```

Comment:
In the code `exit(cmds, sig)`  is followed by `Shutdown(exitCode)`, so `exit` won't ever be called more than once, since `Shutdown` blocks until the sidecar process terminates.